azurerm/internal/services/eventgrid/eventgrid_system_topic_event_subscription_resource.go

```diff
@@ -0,0 +1,738 @@
+				ValidateFunc:  azure.ValidateResourceID,
+			},
+
+			"service_bus_topic_endpoint_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: utils.RemoveFromStringArray(systemTopicEventSubscriptionEndpointPropertyNames(), "service_bus_topic_endpoint_id"),
+				ValidateFunc:  azure.ValidateResourceID,
+			},
+
+			"storage_queue_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: utils.RemoveFromStringArray(systemTopicEventSubscriptionEndpointPropertyNames(), "storage_queue_endpoint"),
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"storage_account_id": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: azure.ValidateResourceID,
+						},
+						"queue_name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+					},
+				},
+			},
+
+			"webhook_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: utils.RemoveFromStringArray(systemTopicEventSubscriptionEndpointPropertyNames(), "webhook_endpoint"),
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"url": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.IsURLWithHTTPS,
+						},
+						"base_url": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"max_events_per_batch": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(1, 5000),
+						},
+						"preferred_batch_size_in_kilobytes": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							ValidateFunc: validation.IntBetween(1, 1024),
+						},
+						"active_directory_tenant_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"active_directory_app_id_or_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"included_event_types": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+			},
+
+			"subject_filter": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"subject_begins_with": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"subject_ends_with": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"case_sensitive": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"advanced_filter": {
```

Comment:
Many of these attributes mirror `azurerm_event_subscription` so we should pull them out of both resources put them into their own helper file since they mirror each other and use the same expand/flatten functions. It'll help quite a bit with maintainability in case they change.

Reply:
Yes. You're right. I try to extract as much as possible from both sources into a helper file.

Reply:
Done. Let me know if this extraction is sufficient for you.